llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -466,6 +468,44 @@ std::string saveModuleProperties(module_split::ModuleDesc &MD,
   return SCFile;
 }
 
+std::string getOptString(module_split::ModuleDesc &MD) {
+  auto &M = MD.getModule();
+  // Process all properties on kernels.
+  for (Function &F : M) {
+    // Only consider kernels.
+    if (F.getCallingConv() != CallingConv::SPIR_KERNEL)
+      continue;
+
+    SmallVector<Metadata *, 8> MDOps;
+    SmallVector<std::pair<std::string, MDNode *>, 8> NamedMDOps;
+    for (const Attribute &Attr : F.getAttributes().getFnAttrs()) {
```

Comment:
If you don't plan to expand this to other attributes, then I would suggest to have a simple pair of `hasFnAttribute` + `getFnAttribute`.

Reply:
Done.